pubsub/rabbitmq/metadata.go

```diff
@@ -97,6 +101,14 @@ func createMetadata(pubSubMetadata pubsub.Metadata, log logger.Logger) (*metadat
 		log.Warn("[DEPRECATION NOTICE] The 'host' argument is deprecated. Use 'connectionString' or individual connection arguments instead: https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-rabbitmq/")
 	}
 
+	if result.connectionString != "" {
+		uri, err := amqp.ParseURI(result.connectionString)
+		if err != nil {
+			return &result, fmt.Errorf("%s invalid connection string: %s, err: %w", errorMessagePrefix, result.connectionString, err)
+		}
+		result.protocol = uri.Scheme
+	}
+
 	if val, found := pubSubMetadata.Properties[metadataProtocolKey]; found && val != "" {
 		result.protocol = val
 	}
```

Comment:
Not sure if I need to check that the protocol parsed from the uri is the same as the configured protocol. If not, raised an error?

Reply:
I don't think that's necessary. The protocol metadata key can be an explicit override for users who know what protocol they want to use.